spec/mercure.md

```diff
@@ -785,14 +687,142 @@ The JSON-LD context available at `https://mercure.rocks` is the following:
+All these attributes are optional.
+
+Minimal example:
+
+~~~ http
+GET /books/foo HTTP/1.1
+Host: example.com
+
+HTTP/1.1 200 OK
+Content-type: application/ld+json
+Link: <https://example.com/.well-known/mercure>; rel="mercure"
+
+{"@id": "/books/foo", "foo": "bar"}
+~~~
+
+## Topic Discovery
+
+The discovery mechanism **MAY** also be used to identify the canonical URL for the topic to which
+subscribers are expected to use for subscriptions.
+
+The publisher **MAY** include one Link Header [@!RFC5988] with `rel=self` (the self link
+header). It **SHOULD** contain the canonical URL for the topic to which subscribers are expected
+to use for subscriptions. If the Link with `rel=self` is omitted, the current URL of the resource
+**MAY** be used as a fallback.
+
+Links embedded in HTML or XML documents as defined in the WebSub recommendation
+[@W3C.REC-websub-20180123] **MAY** also be supported by subscribers. If both a header and an
+embedded link are provided, the header **MUST** be preferred.
+
+### Content Negotiation
+
+For practical purposes, it is important that the `rel=self` URL only offers a single representation.
+As the hub has no way of knowing what Media Type ([@RFC6838]) or language may have been requested
+by the subscriber upon discovery, it would not be able to deliver the content using the appropriate
+representation of the document.
+
+It is, however, possible to perform content negotiation by returning an appropriate `rel=self`
+URL according to the HTTP headers used in the initial discovery request. For example, a request
+to `/books/foo` with an `Accept` header containing `application/ld+json` could return a `rel=self`
+value of `/books/foo.jsonld`.
+
+The example below illustrates how a topic URL can return different `Link` headers depending on the
+`Accept` header that was sent.
+
+~~~ http
+GET /books/foo HTTP/1.1
+Host: example.com
+Accept: application/ld+json
+
+HTTP/1.1 200 OK
+Content-type: application/ld+json
+Link: </books/foo.jsonld>; rel="self"
+Link: <https://example.com/.well-known/mercure>; rel="mercure"
+
+{"@id": "/books/foo", "foo": "bar"}
+~~~
+
+~~~ http
+GET /books/foo HTTP/1.1
+Host: example.com
+Accept: text/html
+
+HTTP/1.1 200 OK
+Content-type: text/html
+Link: </books/foo.html>; rel="self"
+Link: <https://example.com/.well-known/mercure>; rel="mercure"
+
+<!doctype html>
+<link rel="self" href="/books/foo.html">
+<link rel="mercure" href="https://example.com/.well-known/mercure">
+<title>foo: bar</title>
+~~~
+
+Similarly, the technique can also be used to return a different `rel=self` URL depending on the
+language requested by the `Accept-Language` header.
+
+~~~ http
+GET /books/foo HTTP/1.1
+Host: example.com
+Accept-Language: fr-FR
+
+HTTP/1.1 200 OK
+Content-type: application/ld+json
+Content-Language: fr-FR
+Link: </books/foo-fr-FR.jsonld>; rel="self"
+Link: <https://example.com/.well-known/mercure>; rel="mercure"
+
+{"@id": "/books/foo", "foo": "bar", "@context": {"@language": "fr-FR"}}
+~~~
```

Comment:
nothing has changed there it just got moved?

Reply:
yes